src/x509.c

```diff
@@ -7202,6 +7214,402 @@ WOLFSSL_X509_CRL* wolfSSL_d2i_X509_CRL(WOLFSSL_X509_CRL** crl,
+ * wolfSSL_X509_CRL_print()
+ * return WOLFSSL_SUCCESS on success
+ */
+static int X509CRLPrintExtensions(WOLFSSL_BIO* bio, WOLFSSL_X509_CRL* crl,
+        int indent)
+{
+    char tmp[MAX_WIDTH]; /* buffer for XSNPRINTF */
+
+    XSNPRINTF(tmp, MAX_WIDTH, "%*s%s\n", indent, "",
+            "CRL extensions:");
+
+    if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+    }
+
+    if (crl->crlList->crlNumber) {
+        XSNPRINTF(tmp, MAX_WIDTH, "%*s%s\n", indent + 4, "",
+            "X509v3 CRL Number:");
+
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+
+        if (XSNPRINTF(tmp, MAX_WIDTH, "%*s%d\n", indent + 8, "",
+            crl->crlList->crlNumber) >= MAX_WIDTH)
+        {
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        XMEMSET(tmp, 0, sizeof(tmp));
+    }
+
+#if !defined(NO_SKID)
+    if (crl->crlList->extAuthKeyIdSet && crl->crlList->extAuthKeyId[0] != 0) {
+        word32 i;
+        char val[5];
+        int valSz = 5;
+
+        XSNPRINTF(tmp, MAX_WIDTH, "%*s%s", indent + 4, "",
+            "X509v3 Authority Key Identifier:");
+
+        XSTRNCAT(tmp, "\n", MAX_WIDTH - XSTRLEN(tmp) - 1);
+
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        XMEMSET(tmp, 0, MAX_WIDTH);
+
+        XSNPRINTF(tmp, MAX_WIDTH - 1, "%*s%s",
+                        indent + 8, "", "keyid");
+
+        for (i = 0; i < XSTRLEN((char*)crl->crlList->extAuthKeyId); i++) {
+            /* check if buffer is almost full */
+            if (XSTRLEN(tmp) >= sizeof(tmp) - valSz) {
+                if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+                    return WOLFSSL_FAILURE;
+                }
+                tmp[0] = '\0';
+            }
+            if (XSNPRINTF(val, valSz, ":%02X", crl->crlList->extAuthKeyId[i])
+                >= valSz)
+            {
+                WOLFSSL_MSG("buffer overrun");
+                return WOLFSSL_FAILURE;
+            }
+            XSTRNCAT(tmp, val, valSz);
+        }
+        XSTRNCAT(tmp, "\n", XSTRLEN("\n") + 1);
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
+#endif
+
+    return WOLFSSL_SUCCESS;
+}
+
+/* iterate through a CRL's Revoked Certs and print out in human
+ * readable format for use with wolfSSL_X509_CRL_print()
+ * return WOLFSSL_SUCCESS on success
+ */
+static int X509CRLPrintRevoked(WOLFSSL_BIO* bio, WOLFSSL_X509_CRL* crl,
+        int indent)
+{
+    char tmp[MAX_WIDTH]; /* buffer for XSNPRINTF */
+
+    if (crl->crlList->totalCerts > 0) {
+        RevokedCert* revoked = crl->crlList->certs;
+
+        XSNPRINTF(tmp, MAX_WIDTH, "%*s%s\n", indent, "",
```

Comment:
check on XSNPRINTF return value

Reply:
please grep through all XSNPRINTF's in this PR to check the return values

Reply:
extended testing that is done after the merge would complain about the ret value (see https://github.com/wolfSSL/wolfssl/pull/5445)